code/datums/status_effects/debuffs/heart_attack.dm

```diff
@@ -0,0 +1,162 @@
+
+	if(time_until_stoppage > ATTACK_STAGE_THREE)
+		if(prob(5))
+			owner.playsound_local(owner, 'sound/effects/singlebeat.ogg', 25, FALSE, use_reverb = FALSE)
+			owner.adjustStaminaLoss(5, FALSE)
+
+	if(time_until_stoppage <= ATTACK_STAGE_TWO && time_until_stoppage > ATTACK_STAGE_THREE)	//This coughing gets replaced with worse coughing, no need to stack it.
+		owner.playsound_local(owner, 'sound/effects/health/slowbeat.ogg', 25, FALSE, channel = CHANNEL_HEARTBEAT, use_reverb = FALSE)
+		if(prob(10))
+			owner.emote("cough")
+			owner.adjustStaminaLoss(10)
+			oxyloss_sum += 4
+
+	if(time_until_stoppage <= ATTACK_STAGE_THREE) //At this point, we start with chat messages and make it clear that something is very wrong.
+		if(!visible)
+			ADD_TRAIT(owner, TRAIT_DISEASELIKE_SEVERITY_HIGH, type)
+			owner.med_hud_set_status()
+			visible = TRUE //We do not reset this status until it's fully cured. Once it's been made apparent, there's no reason to hide it again until it is resolved. It will only confuse players.
+		if(prob(15))
+			to_chat(owner, span_danger("You feel a sharp pain in your chest!"))
+			if(prob(25))
+				human_owner.vomit(VOMIT_CATEGORY_DEFAULT, lost_nutrition = 95)
+			owner.emote("cough")
+		if(prob(8))
+			to_chat(owner, span_danger("You feel very weak and dizzy..."))
+			owner.adjust_confusion_up_to(6 SECONDS, 10 SECONDS)
+			owner.adjustStaminaLoss(20)
+			owner.emote("cough")
+			oxyloss_sum += 8
+
+	if(time_until_stoppage <= ATTACK_STAGE_FOUR) //And now we compound it with even worse effects.
+		owner.stop_sound_channel(CHANNEL_HEARTBEAT)
+		sound = FALSE
+		owner.playsound_local(owner, 'sound/effects/singlebeat.ogg', 100, FALSE, use_reverb = FALSE)
+
+		if(prob(5))
+			to_chat(owner, span_userdanger("It feels like you're shutting down..."))
+			owner.adjust_dizzy_up_to(4 SECONDS, 10 SECONDS)
+			owner.adjust_eye_blur_up_to(4 SECONDS, 20 SECONDS)
+			owner.adjustStaminaLoss(20)
+
+		if(prob(5))
+			owner.emote("cough")
+			if(prob(5))
+				to_chat(owner, span_userdanger("You cough. Everything goes dark. You're going to die soon."))
+				owner.adjust_temp_blindness(10 SECONDS) //Are you panicking yet? You should be panicking by now.
+			else
+				to_chat(owner, span_userdanger("As you cough, your chest surges in pain and darkness closes in around your sight."))
+				owner.adjust_temp_blindness(2 SECONDS)
+				owner.adjust_eye_blur_up_to(4 SECONDS, 20 SECONDS)
+			oxyloss_sum += 8
+			owner.Paralyze(10)
+		oxyloss_sum += 3
+
+	if(owner.getOxyLoss() < OXYLOSS_MAXIMUM) //A bad enough roll on the verge of passing out might still push you over into unconciousness for a few seconds...?
+		owner.adjustOxyLoss(oxyloss_sum)
+
+	if(time_until_stoppage <= 0)
+		if(owner.stat == CONSCIOUS)
+			owner.visible_message(span_danger("[owner] clutches at [owner.p_their()] chest as if [owner.p_their()] heart is stopping!"), \
+			span_userdanger("You feel a terrible pain in your chest, as if your heart has stopped!"))
```

Comment:
```suggestion
			owner.visible_message(
				span_danger("[owner] clutches at [owner.p_their()] chest as if [owner.p_their()] heart is stopping!"),
				span_userdanger("You feel a terrible pain in your chest, as if your heart has stopped!"), \
			)
```

Reply:
ended up swapping it to a to_chat since the message to the victim is sent in the heart's `on_life()`